proc/process.c

```diff
@@ -1096,15 +1098,31 @@ static void process_exec(thread_t *current, process_spawn_t *spawn)
 #ifndef NOMMU
 	vm_mapCreate(&current->process->map, (void *)(VADDR_MIN + SIZE_PAGE), (void *)VADDR_USR_MAX);
 	proc_changeMap(current->process, &current->process->map, NULL, &current->process->map.pmap);
+	(void)i;
 #else
+	pmap_create(&current->process->map.pmap, NULL, NULL, NULL);
 	proc_changeMap(current->process, (spawn->map != NULL) ? spawn->map : process_common.kmap, spawn->imap, &current->process->map.pmap);
-
 	current->process->entries = NULL;
+
+	if (spawn->prog != NULL) {
+		/* Add instruction maps */
+		for (i = 0; (i < spawn->prog->imapSz) && (err >= 0); ++i) {
+			err = pmap_addMap(current->process->pmapp, spawn->prog->imaps[i]);
+		}
+
+		/* Add data/io maps */
+		for (i = 0; (i < spawn->prog->dmapSz) && (err >= 0); ++i) {
+			err = pmap_addMap(current->process->pmapp, spawn->prog->dmaps[i]);
+		}
+	}
 #endif
 
 	pmap_switch(current->process->pmapp);
 
-	err = process_load(current->process, spawn->object, spawn->offset, spawn->size, &stack, &entry);
+	if (err <= 0) {
+		err = process_load(current->process, spawn->object, spawn->offset, spawn->size, &stack, &entry);
+	}
```

Comment:
Oh, and one more thing - from clang diagnostic, `entry` is used not initialized when `err > 0`. This won't happen with current code path as `pmap_addMap` returns `<= 0` so maybe it's ok to remove this `if`?

Reply:
Ugh, this condition was a major brain-fart, fixed. I also fixed this potential usage of not initialized `entry`